File: libs/resea/task.c
#include <resea/syscall.h>
#include <resea/task.h>

task_t task_create(const char *name, vaddr_t ip, task_t pager, unsigned flags) {
    return sys_task_create(name, ip, pager, flags);
}

error_t task_destroy(task_t task) {
    return sys_task_destroy(task);
}

__noreturn void task_exit(void) {
    sys_task_exit();
    for (;;)
        ;
}

task_t task_self(void) {
    return sys_task_self();
}

File: libs/resea/rust/src/task.rs
use crate::capi::task_t;

pub struct Task {
    id: task_t,
}

impl Task {
    pub unsafe fn from_id(id: task_t) -> Task {
        Task { id }
    }

    pub fn id(&self) -> task_t {
        self.id
    }
}
